src/rules/ResourceQuoteStyle.js

```diff
@@ -64,155 +119,193 @@ class ResourceQuoteStyle extends ResourceRule {
-            re2 = new RegExp(`([\\p{Letter}\\}])([${regExps.target.nonQuoteCharsAlt}'"])(\\W|$)`, "gu");
+        let highlight, description, lineNumber;
+        if (startQuote.test(tar) || endQuote.test(tar)) {
+            highlight = tar;
+            if (startQuote) { highlight = highlight.replace(startQuote, "$1<e0>$2</e0>$3"); }
+            if (endQuote) { highlight = highlight.replace(endQuote, "$1<e1>$2</e1>$3"); }
+            highlight = `Target: ${highlight}`;
+            description = `Quote style for the locale ${locale} should be ${targetQuoteStyleExample}`;
+        } else {
+            highlight = `Target: ${tar}<e0></e0>`;
+            description = `Quotes are missing in the target. Quote style for the locale ${locale} should be ${targetQuoteStyleExample}`;
         }
-        const matches1 = re1 ? re1.exec(tar) : undefined;
-        const matches2 = re2 ? re2.exec(tar) : undefined;
+        // @ts-ignore: Property 'lineNumber' does not exist on type 'Resource'
+        // there is no lineNumber property on a Resource type
+        // (preserved for compatibility)
+        if (typeof(resource.lineNumber) !== 'undefined') {lineNumber = /** @type {number} */ (resource.lineNumber); }
 
-        let value = {
-            severity: this.localeOnly ? "error" : "warning",
+        return new Result({
+            /** @type {Severity} */ severity: this.localeOnly ? "error" : "warning",
             id: resource.getKey(),
             source: src,
             rule: this,
             locale,
-            pathName: file
-        };
-        if (matches1 || matches2) {
-            value.highlight = `Target: ${tar.replace(re1, "$1<e0>$2</e0>$3").replace(re2, "$1<e1>$2</e1>$3")}`;
-            value.description = `Quote style for the locale ${locale} should be ${quoteStyle}`;
-        } else {
-            value.highlight = `Target: ${tar}<e0></e0>`;
-            value.description = `Quotes are missing in the target. Quote style for the locale ${locale} should be ${regExps.target.quoteStart}text${regExps.target.quoteEnd}`;
-        }
-        if (typeof(resource.lineNumber) !== 'undefined') {
-            value.lineNumber = resource.lineNumber;
-        }
-        return new Result(value);
+            pathName: file,
+            highlight,
+            description,
+            lineNumber,
+        });
     }
 
     /**
      * Calculate all the regular expressions we need.
      * @private
+     * @param {string} locale
+     * @returns {RegExpCollectionForLocale}
      */
     getRegExps(locale) {
-        if (regExpsCache[locale]) return regExpsCache[locale];
-
-        // store quote chars and regexps in here and then cache it
-        // so we only do it once for each locale
-        let regExps = {
-            source: {},
-            target: {}
-        };
-
+        // superset of all the non-ASCII start and end chars used in CLDR
+        const quoteChars = "«»‘“”„「」’‚‹›『』";
```

Comment:
If you define a variable inside of a function/method, then it defines it each time the function/method is called. Static constant variables should stay outside of the functions/method so that they are only defined once.

Reply:
Performance-wise I'm pretty sure that it has practically zero impact, V8 optimises this kind of stuff at runtime.

I moved it here because it's not used anywhere else other than during creation of per-locale regexps (L259 and L264)

Reply:
ok